wallet/wallet.c

```diff
@@ -281,6 +281,15 @@ bool wallet_update_output_status(struct wallet *w,
 {
 	struct db_stmt *stmt;
 	size_t changes;
+
+	/* FIXME(vincenzopalazzo): There are different nasty case at this point that are
+	 *
+	 * - moving from OUTPUT_STATE_SPENT -> OUTPUT_STATE_CONFIRMENT required
+	 * to set the spendheight to null
+	 * - moving from OUTPUT_STATE_CONFIRMED -> OUTPUT_STATE_SPENT required
+	 * to set the spendheight.
+	 *
+	 * in both bases the following code do not do that. */
```

Comment:
Shall we encapsulate the status update logic into a function (like you did further down), and then force all updates to go through that logic? It could even have an assert that ensures the `enum -> spendheight?` mapping, enforcing that marking as spent sets, and marking as unspent unsets.

Reply:
How we can make the transition to a spending state if we do not have an API that has the `spendheight` information?